pkg/docker/docker-entrypoint.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+set -e
+
+# first arg is a flag.
+if [ "${1:0:1}" = '-' ]; then
```

Comment:
What if there isn't a first argument?

Reply:
I'm not sure what you mean. Please see if the latest update addresses your concerns.

Reply:
I was worried that a missing `$1` would cause it to error out when `nounset` is in effect, but that seems to be ignored for substring substitutions (i.e., `echo "${1}"` would produce a "1: unbound variable" error, but `echo ${1:0:1}` will happily print an empty string). So never mind.

Reply:
@bmoyles0117 Just noticed that this is a `bash`-ism, and you set the shell to `/bin/sh`. I got `Bad substitution` errors on this line in the latest container.